File: 백준/Gold/1644. 소수의 연속합/소수의 연속합.py
import math
N = int(input())
arr = [1] * (N+1)
for i in range(2, int(math.sqrt(N))+1):
    if arr[i]:
        for j in range(2*i, N+1, i):
            arr[j] = 0
            
prime = []
for i in range(2, len(arr)):
    if arr[i]:
        prime.append(i)
        
end = ans = hap = 0
for start in range(len(prime)):

    while hap < N and end < len(prime):
        hap += prime[end]
        end += 1

    if hap == N:
        ans += 1
    hap -= prime[start]

print(ans)

File: 백준/Gold/1644. 소수의 연속합/소수의 연속합.java
import java.awt.List;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
public class Main{
	public static void main (String [] args) throws IOException {
	BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	int N = Integer.parseInt(br.readLine());
	int[] arr = new int [N+1];
	for (int i = 2; i < (int)Math.sqrt(N)+1; i++) {
		if (arr[i] == 0) {
			for (int j = i*2; j < arr.length; j+=i) {
				arr[j] = 1;
			}
		}
	}
	ArrayList<Integer> prime = new ArrayList<>();
	for (int i = 2; i < arr.length; i++) {
		if (arr[i] == 0) {
			prime.add(i);
		}
	}
	int end = 0;
	int cnt = 0;
	int hap = 0;
	for (int i = 0; i < prime.size(); i++) {
		
		while (hap < N && end < prime.size()) {
			hap += prime.get(end);
			end += 1;			
		}
		if (hap == N) {
			cnt += 1;
		}	
		hap -= prime.get(i);

	}
	System.out.println(cnt);
	}
}